### PR TITLE
Add the docker-distribution snap

### PR DESCRIPTION
**Pull request by come-maiz**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/ubuntu/snappy-playpen/pull/198%23discussion_r72390788%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/198%23discussion_r72390867%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/198%23discussion_r72390902%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/198%23issuecomment-235506226%22%2C%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/198%23issuecomment-237436410%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/198%23issuecomment-235506226%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Great%20work%21%20This%20looks%20good%20to%20me%20%3A%29%5Cr%5Cn%5Cr%5CnAre%20you%20going%20to%20continue%20on%20this%20to%20try%20removing%20devmode%20before%20we%20merge%3F%20I%20know%20%40LyzardKing%20worked%20on%20some%20docker%20snap%20as%20well%2C%20maybe%20you%20can%20both%20work%20together%20to%20get%20that%20working%3F%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222016-07-27T07%3A19%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1823296%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/didrocks%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20pushed%20all%20your%20suggested%20changes%20%40didrocks.%20Thanks%20for%20the%20review.%5Cr%5CnThe%20direction%20I%20got%20was%20to%20write%20as%20many%20snaps%20as%20possible%20in%20devmode%2C%20and%20then%20try%20to%20get%20them%20to%20strict%2C%20so%20I%20wasn%27t%20planning%20to%20spend%20much%20more%20time%20on%20this%20one%20on%20the%20short%20term.%20I%20will%20revisit%20it%20when%20I%20have%20some%20free%20time.%20So%20I%20would%20say%20that%27s%20better%20to%20get%20it%20merged%2C%20but%20I%27m%20also%20ok%20with%20leaving%20it%20in%20a%20PR.%22%2C%20%22created_at%22%3A%20%222016-08-04T02%3A36%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/617831%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/elopio%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20306541b990a98bcdb641e41a89a28655e0a86558%20docker-distribution/snapcraft.yaml%2020%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/198%23discussion_r72390867%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22We%20tend%20to%20set%20them%20in%20that%20order%3A%5Cr%5Cn%60%60%60%5Cr%5Cnsource%3A%5Cr%5Cnsource-options%3A%5Cr%5Cnplugins%3A%5Cr%5Cnplugins-options%3A%5Cr%5Cn%60%60%60%5Cr%5CnMind%20reordering%20this%20to%20follow%20those%3F%22%2C%20%22created_at%22%3A%20%222016-07-27T07%3A17%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1823296%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/didrocks%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20docker-distribution/snapcraft.yaml%3AL1-25%22%7D%2C%20%22Pull%20306541b990a98bcdb641e41a89a28655e0a86558%20docker-distribution/snapcraft.yaml%2022%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/198%23discussion_r72390902%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Please%20add%20.git%20to%20%60source%60%20and%20remove%20%60source-type%60%20then.%22%2C%20%22created_at%22%3A%20%222016-07-27T07%3A17%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1823296%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/didrocks%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20docker-distribution/snapcraft.yaml%3AL1-25%22%7D%2C%20%22Pull%20306541b990a98bcdb641e41a89a28655e0a86558%20docker-distribution/snapcraft.yaml%202%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ubuntu/snappy-playpen/pull/198%23discussion_r72390788%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Please%20enclose%20this%20with%20%5C%22%5C%22%20to%20force%20a%20string%20%28to%20follow%20our%20conventions%29%22%2C%20%22created_at%22%3A%20%222016-07-27T07%3A16%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1823296%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/didrocks%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20docker-distribution/snapcraft.yaml%3AL1-25%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull 306541b990a98bcdb641e41a89a28655e0a86558 docker-distribution/snapcraft.yaml 2'></a> &nbsp;<b><a href='https://github.com/ubuntu/snappy-playpen/pull/198#discussion_r72390788'>File: docker-distribution/snapcraft.yaml:L1-25</a></b>
- <a href='https://github.com/didrocks'></a> Please enclose this with "" to force a string (to follow our conventions)
- [x] <a href='#crh-comment-Pull 306541b990a98bcdb641e41a89a28655e0a86558 docker-distribution/snapcraft.yaml 20'></a> &nbsp;<b><a href='https://github.com/ubuntu/snappy-playpen/pull/198#discussion_r72390867'>File: docker-distribution/snapcraft.yaml:L1-25</a></b>
- <a href='https://github.com/didrocks'></a> We tend to set them in that order:

```
source:
source-options:
plugins:
plugins-options:
```

Mind reordering this to follow those?
- [x] <a href='#crh-comment-Pull 306541b990a98bcdb641e41a89a28655e0a86558 docker-distribution/snapcraft.yaml 22'></a> &nbsp;<b><a href='https://github.com/ubuntu/snappy-playpen/pull/198#discussion_r72390902'>File: docker-distribution/snapcraft.yaml:L1-25</a></b>
- <a href='https://github.com/didrocks'></a> Please add .git to `source` and remove `source-type` then.
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/ubuntu/snappy-playpen/pull/198#issuecomment-235506226'>General Comment</a></b>
- <a href='https://github.com/didrocks'></a> Great work! This looks good to me :)
  Are you going to continue on this to try removing devmode before we merge? I know @LyzardKing worked on some docker snap as well, maybe you can both work together to get that working?
- <a href='https://github.com/elopio'></a> I pushed all your suggested changes @didrocks. Thanks for the review.
  The direction I got was to write as many snaps as possible in devmode, and then try to get them to strict, so I wasn't planning to spend much more time on this one on the short term. I will revisit it when I have some free time. So I would say that's better to get it merged, but I'm also ok with leaving it in a PR.

<a href='https://www.codereviewhub.com/ubuntu/snappy-playpen/pull/198?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/ubuntu/snappy-playpen/pull/198?approve=1'></a>&nbsp;<a href='https://github.com/ubuntu/snappy-playpen/pull/198'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by didrocks** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/ubuntu/snappy-playpen/pull/198'>Refresh</a>.<a href='#crh-update-none'></a>

File: docker-distribution/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: docker-distribution
+version: v2.5.0-rc.2
```
Comment (didrocks) [inline docker-distribution/snapcraft.yaml]: Please enclose this with "" to force a string (to follow our conventions)

File: docker-distribution/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  digest:
+    command: bin/digest
+  registry:
+    command: bin/registry
+  registry-api-descriptor-template:
+    command: bin/registry-api-descriptor-template
+
+parts:
+  distribution:
+    plugin: distribution
```
Comment (didrocks) [inline docker-distribution/snapcraft.yaml]: We tend to set them in that order:
```
source:
source-options:
plugins:
plugins-options:
```
Mind reordering this to follow those?

File: docker-distribution/snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    source: https://github.com/docker/distribution
+    source-type: git
```
Comment (didrocks) [inline docker-distribution/snapcraft.yaml]: Please add .git to `source` and remove `source-type` then.

**Comment by didrocks** [comment]
Great work! This looks good to me :)

Are you going to continue on this to try removing devmode before we merge? I know @LyzardKing worked on some docker snap as well, maybe you can both work together to get that working?

**Comment by elopio** [comment]
I pushed all your suggested changes @didrocks. Thanks for the review.
The direction I got was to write as many snaps as possible in devmode, and then try to get them to strict, so I wasn't planning to spend much more time on this one on the short term. I will revisit it when I have some free time. So I would say that's better to get it merged, but I'm also ok with leaving it in a PR.

**Pull request closed without merging**